src/main/drivers/usb_msc_f7xx.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Chris Hockuba (https://github.com/conkerkh)
+ *
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "platform.h"
+
+#if defined(USE_USB_MSC)
+
+#include "build/build_config.h"
+
+#include "common/utils.h"
+
+#include "drivers/io.h"
+#include "drivers/light_led.h"
+#include "drivers/nvic.h"
+#include "drivers/time.h"
+#include "drivers/usb_msc.h"
+
+#include "pg/usb.h"
+
+#include "vcp_hal/usbd_cdc_interface.h"
+#include "usb_io.h"
+#include "usbd_msc.h"
+#include "msc/usbd_storage.h"
+USBD_HandleTypeDef USBD_Device;
+
+#define DEBOUNCE_TIME_MS 20
+
+static IO_t mscButton;
+
+void mscInit(void)
+{
+    if (usbDevConfig()->mscButtonPin) {
+        mscButton = IOGetByTag(usbDevConfig()->mscButtonPin);
+        IOInit(mscButton, OWNER_USB_MSC_PIN, 0);
+        if (usbDevConfig()->mscButtonUsePullup) {
+            IOConfigGPIO(mscButton, IOCFG_IPU);
+        } else {
+            IOConfigGPIO(mscButton, IOCFG_IPD);
+        }
+    }
+}
+
+uint8_t mscStart(void)
+{
+	ledInit(statusLedConfig());
+
+	//Start USB
+	usbGenerateDisconnectPulse();
+
+	IOInit(IOGetByTag(IO_TAG(PA11)), OWNER_USB, 0);
+	IOInit(IOGetByTag(IO_TAG(PA12)), OWNER_USB, 0);
+
+	USBD_Init(&USBD_Device, &VCP_Desc, 0);
+
+	/** Regsiter class */
+	USBD_RegisterClass(&USBD_Device, USBD_MSC_CLASS);
+
+	/** Register interface callbacks */
+	USBD_MSC_RegisterStorage(&USBD_Device, &USBD_MSC_Template_fops);
+
+	USBD_Start(&USBD_Device);
+
+	// NVIC configuration for SYSTick
+	NVIC_DisableIRQ(SysTick_IRQn);
+	NVIC_SetPriority(SysTick_IRQn, NVIC_BUILD_PRIORITY(0, 0));
+	NVIC_EnableIRQ(SysTick_IRQn);
+
+	return 0;
+}
+
+bool mscBoot(void)
```

Comment:
This function name doesn't really state what it does. `mscCheckBoot()`?

Reply:
yep I will rename